Assets/Scripts/Model/Weapon.cs

```diff
@@ -32,13 +40,46 @@ void Update ()
 
 	}
 
+	// Upgrade weapon features
 	public void Upgrade ()
 	{
 		this.currentDamage *= upgradeFactor;
 	}
+		
+	//USeed for sounds
+	void Awake()
+	{
+		this.source = GetComponent<AudioSource>();
+	}
 
-    /*public float getDamage()
+	// Getters
+    public float getCurrentDamage()
     {
         return this.currentDamage;
-    }*/
+    }
+
+	public void addTarget(CanReceiveDamage target){
+		this.targets.Add (target);
+		Debug.Log ("Target in queue");
+	}
+
+	// Called to attack a target
+	public void Attack()
```

Comment:
While I was writing a response I've realized that your solution might be even better than what I've thought initially xd nice work!